pkg/scm/git/git_test.go

```diff
@@ -13,23 +13,56 @@ import (
 	"github.com/openshift/source-to-image/pkg/test"
 )
 
+// Creates a git directory with one unlikely but possible commit hash
 func createLocalGitDirectory(t *testing.T) string {
-	dir, err := ioutil.TempDir(os.TempDir(), "s2i-test")
+	dir, err := ioutil.TempDir(os.TempDir(), "gitdir-s2i-test")
 	if err != nil {
 		t.Error(err)
 	}
-	os.Mkdir(filepath.Join(dir, ".git"), 0600)
+	os.MkdirAll(filepath.Join(dir, ".git/refs/heads"), 0777)
+	os.MkdirAll(filepath.Join(dir, ".git/refs/remotes"), 0777)
+	os.MkdirAll(filepath.Join(dir, ".git/branches"), 0777)
+	os.MkdirAll(filepath.Join(dir, ".git/objects/fo"), 0777)
+	os.Create(filepath.Join(dir, ".git/objects/fo") + "12345678901234567890123456789012345678") // 40 character SHA-1 hash
 	return dir
 }
 
-func TestIsLocalGitRepository(t *testing.T) {
+func createEmptyLocalGitDirectory(t *testing.T) string {
+	dir, err := ioutil.TempDir(os.TempDir(), "gitdir-s2i-test")
+	if err != nil {
+		t.Error(err)
+	}
+	os.MkdirAll(filepath.Join(dir, ".git/refs/heads"), 0777)
+	os.MkdirAll(filepath.Join(dir, ".git/refs/remotes"), 0777)
+	os.MkdirAll(filepath.Join(dir, ".git/branches"), 0777)
+	os.MkdirAll(filepath.Join(dir, ".git/objects"), 0777)
+	return dir
+}
+
+func TestIsValidGitRepository(t *testing.T) {
 	d := createLocalGitDirectory(t)
-	defer os.RemoveAll(d)
-	if isLocalGitRepository(d) == false {
+
+	// We have a .git that is populated
+	// Should return true with no error
+	if ok, err := isValidGitRepository(d); ok == false || err != nil {
 		t.Errorf("The %q directory is git repository", d)
 	}
-	os.RemoveAll(filepath.Join(d, ".git"))
-	if isLocalGitRepository(d) == true {
+
+	os.RemoveAll(d)
+	d = createEmptyLocalGitDirectory(t)
+	defer os.RemoveAll(d)
+
+	// There are no tracking objects in the .git repository
+	// Should return true with an error
+	if ok, err := isValidGitRepository(d); ok == false || err == nil {
+		t.Errorf("The %q directory is not a git repository", d)
```

Comment:
for completeness you should check that the error you got is the type of error (emptygitdir) you expected.